server/container/util/src/main/java/org/apache/james/util/CompletableFutureUtil.java

```diff
@@ -111,4 +112,12 @@
         failedFuture.completeExceptionally(throwable);
         return failedFuture;
     }
+
+    public static <T> BiConsumer<T, Throwable> whenSuccess(Runnable runnable) {
```

Comment:
This feels like `Try` or `Either` monad

Reply:
I don't really get it. Can you provide an example? Idk those class (Try or Either) come from.

Reply:
They are typical monads, you can read about Try and Either in vavr.io documentation or scala documentation